### ISSUE TITLE
add_noise scales by variance, not its root

### ISSUE THREAD

**Issue by Rupt** [deepmind/optax#281]
To match its intended variance, [add_noise](https://github.com/deepmind/optax/blob/5fc395b5e49602e3c0027b976d1368755b20cd0b/optax/_src/transform.py#L725) should scale by its square root.
```diff
-        lambda g, n: g + variance.astype(g.dtype) * n,
+        lambda g, n: g + jnp.sqrt(variance.astype(g.dtype)) * n,
```

I am posting this as an issue rather than a pull request since this is a strongly breaking change which deserves some consideration. Thank you for maintaining and sharing optax!

**Comment by mkunesch** [comment]
Hi @Rupt!

Thank you so much for reporting this! This is indeed a bug and should be fixed. Normally, I'd say we shouldn't quietly change behaviours but I think in this case it's fine (`add_noise` and `noisy_sgd` don't seem to be used very commonly and we are still in the 0.*.* version numbers).

Do you want to file a PR for this (+ a failing test) or shall I? (We'll release a new version after the fix)

Thanks a lot again for reporting!

**Comment by Rupt** [comment]
Thanks @mkunesch for the feedback. This makes sense to me, and I would like to try the PR (with test) now.

**Comment by mkunesch** [comment]
Amazing, thanks a lot! I'll review the code - we might even be able to include it with the version we want to release with the complex number support (#279) this week.

**Comment by mkunesch** [comment]
This was fixed in #294 so I'll close this issue. Thanks @Rupt for spotting and fixing this bug!
